From 159cb7bc58b56fcd59f9541297578fe5d0260327 Mon Sep 17 00:00:00 2001
From: Luc Verhaegen <libv@skynet.be>
Date: Thu, 1 Nov 2012 21:16:20 +0100
Subject: [PATCH 912/944] video:sunxi:disp: make Display_Fb_Release less
 noisy.

Reorder the fb enable checking to immediately return when the fb has
not been enabled. The warning message there is quite useless.

Also, the fb_id checking is pointless, fb_id is always above 0 as it
is a __u32

Signed-off-by: Luc Verhaegen <libv@skynet.be>
---
 drivers/video/sunxi/disp/dev_fb.c | 43 +++++++++++++++++----------------------
 1 file changed, 19 insertions(+), 24 deletions(-)

diff --git a/drivers/video/sunxi/disp/dev_fb.c b/drivers/video/sunxi/disp/dev_fb.c
index 0f3ec47..739b2c6 100644
--- a/drivers/video/sunxi/disp/dev_fb.c
+++ b/drivers/video/sunxi/disp/dev_fb.c
@@ -1344,37 +1344,32 @@ __s32 Display_Fb_Request(__u32 fb_id, __disp_fb_create_para_t * fb_para)
 
 __s32 Display_Fb_Release(__u32 fb_id)
 {
+	struct fb_info *info = g_fbi.fbinfo[fb_id];
+	__u32 sel = 0;
+
 	__inf("Display_Fb_Release, fb_id:%d\n", fb_id);
 
-	if ((fb_id >= 0) && g_fbi.fb_enable[fb_id]) {
-		__u32 sel = 0;
-		struct fb_info *info = g_fbi.fbinfo[fb_id];
+	if (!g_fbi.fb_enable[fb_id])
+		return DIS_SUCCESS;
 
-		for (sel = 0; sel < 2; sel++) {
-			if (((sel == 0) &&
-			     (g_fbi.fb_mode[fb_id] != FB_MODE_SCREEN1)) ||
-			    ((sel == 1) &&
-			     (g_fbi.fb_mode[fb_id] != FB_MODE_SCREEN0))) {
-				__s32 layer_hdl = g_fbi.layer_hdl[fb_id][sel];
+	for (sel = 0; sel < 2; sel++)
+		if (((sel == 0) && (g_fbi.fb_mode[fb_id] != FB_MODE_SCREEN1)) ||
+		    ((sel == 1) && (g_fbi.fb_mode[fb_id] != FB_MODE_SCREEN0))) {
+			__s32 layer_hdl = g_fbi.layer_hdl[fb_id][sel];
 
-				BSP_disp_layer_release(sel, layer_hdl);
-			}
+			BSP_disp_layer_release(sel, layer_hdl);
 		}
-		g_fbi.layer_hdl[fb_id][0] = 0;
-		g_fbi.layer_hdl[fb_id][1] = 0;
-		g_fbi.fb_mode[fb_id] = FB_MODE_SCREEN0;
-		memset(&g_fbi.fb_para[fb_id], 0,
-		       sizeof(__disp_fb_create_para_t));
-		g_fbi.fb_enable[fb_id] = 0;
 
-		fb_dealloc_cmap(&info->cmap);
-		Fb_unmap_video_memory(info);
+	g_fbi.layer_hdl[fb_id][0] = 0;
+	g_fbi.layer_hdl[fb_id][1] = 0;
+	g_fbi.fb_mode[fb_id] = FB_MODE_SCREEN0;
+	memset(&g_fbi.fb_para[fb_id], 0, sizeof(__disp_fb_create_para_t));
+	g_fbi.fb_enable[fb_id] = 0;
 
-		return DIS_SUCCESS;
-	} else {
-		__wrn("invalid paras fb_id:%d in Display_Fb_Release\n", fb_id);
-		return DIS_FAIL;
-	}
+	fb_dealloc_cmap(&info->cmap);
+	Fb_unmap_video_memory(info);
+
+	return DIS_SUCCESS;
 }
 
 __s32 Display_Fb_get_para(__u32 fb_id, __disp_fb_create_para_t *fb_para)
-- 
1.8.0

